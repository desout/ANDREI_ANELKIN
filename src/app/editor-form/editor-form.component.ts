import {Component, OnInit} from '@angular/core';
import {FormControl, FormGroup, Validators} from '@angular/forms';
import {AgeValidator} from '../validators/age.validator';
import {DateValidator} from '../validators/date.validator';
import {createUser, User} from '../../../server/server/User';
import {UserService} from '../user.service';
import {isNameTaken} from '../validators/user-name.validator';

@Component({
  selector: 'app-editor-form',
  templateUrl: './editor-form.component.html',
  styleUrls: ['./editor-form.component.scss']
})
export class EditorFormComponent implements OnInit {
  user = {name: 'Desout',
    age: '18',
    dateOfBirth: '2018/11/11',
    dateOfFirstLogin: '2018/11/11',
    dateNextNotification: '2018/11/11',
    information: 'infoinfo'};
  userForm: FormGroup;
  submitted = false;
  constructor(private userService: UserService) {}

  ngOnInit(): void {
    this.userForm = new FormGroup({
      name: new FormControl(this.user.name, {
        validators: [Validators.required, Validators.minLength(4)],
        asyncValidators: [isNameTaken.bind(this)],
        updateOn: 'blur'
      }),
      age: new FormControl(this.user.age, [Validators.required, AgeValidator]),
      dateOfBirth: new FormControl(this.user.dateOfBirth, [Validators.required, DateValidator]),
      dateOfFirstLogin: new FormControl(this.user.dateOfFirstLogin, [Validators.required, DateValidator]),
      dateNextNotification: new FormControl(this.user.dateNextNotification, [Validators.required, DateValidator]),
      information: new FormControl(this.user.information, [Validators.required, Validators.minLength(4)])

  });
  }
  onSubmit() {
    this.submitted = true;
    const exportUser: User = this.getUserFromForm();
    this.userService.addUser(exportUser).subscribe();
  }
  getUserFromForm(): User {
    return createUser(this.name.value,
      'empty',
      this.dateOfBirth.value,
      this.dateOfFirstLogin.value,
      this.dateNextNotification.value,
      this.information.value,
      'user');
  }
  get name() { return this.userForm.get('name'); }
  get age() { return this.userForm.get('age'); }
  get dateOfBirth() { return this.userForm.get('dateOfBirth'); }
  get dateOfFirstLogin() { return this.userForm.get('dateOfFirstLogin'); }
  get dateNextNotification() { return this.userForm.get('dateNextNotification'); }
  get information() { return this.userForm.get('information'); }
}

