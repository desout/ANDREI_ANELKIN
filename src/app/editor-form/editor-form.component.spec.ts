import { async, ComponentFixture, TestBed } from '@angular/core/testing';

import { EditorFormComponent } from './editor-form.component';
import {NO_ERRORS_SCHEMA} from '@angular/core';
import {TestModule} from '../Test.module';

describe('EditorFormComponent', () => {
  let component: EditorFormComponent;
  let fixture: ComponentFixture<EditorFormComponent>;

  beforeEach(async(() => {
    TestBed.configureTestingModule({
      declarations: [],
      schemas: [NO_ERRORS_SCHEMA],
      imports: [TestModule],
    })
    .compileComponents();
  }));

  beforeEach(() => {
    fixture = TestBed.createComponent(EditorFormComponent);
    component = fixture.componentInstance;
    fixture.detectChanges();
  });

  it('should create', () => {
    expect(fixture.nativeElement).toMatchSnapshot();
  });
});
