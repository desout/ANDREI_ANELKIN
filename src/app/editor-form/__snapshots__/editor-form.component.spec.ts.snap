// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EditorFormComponent should create 1`] = `
<div
  id="root0"
  ng-version="7.0.3"
>
  <div
    class="container"
  >
    <div>
      <form
        class="ng-untouched ng-pristine ng-valid"
        ng-reflect-form="[object Object]"
        novalidate=""
      >
        <div
          class="form-group"
        >
          <label>
            Name 
            <input
              class="form-control ng-untouched ng-pristine ng-valid"
              formcontrolname="name"
              ng-reflect-name="name"
            />
          </label>
          
          
        </div>
        <div
          class="form-group"
        >
          <label>
            age 
            <input
              class="form-control ng-untouched ng-pristine ng-valid"
              formcontrolname="age"
              ng-reflect-name="age"
              ng-reflect-required=""
              required=""
            />
          </label>
          
        </div>
        <div
          class="form-group"
        >
          <label>
            birthday 
            <input
              class="form-control ng-untouched ng-pristine ng-valid"
              formcontrolname="dateOfBirth"
              ng-reflect-name="dateOfBirth"
              ng-reflect-required=""
              required=""
            />
          </label>
          
        </div>
        <div
          class="form-group"
        >
          <label>
            date Of Login 
            <input
              class="form-control ng-untouched ng-pristine ng-valid"
              formcontrolname="dateOfFirstLogin"
              ng-reflect-name="dateOfFirstLogin"
              ng-reflect-required=""
              required=""
            />
          </label>
          
        </div>
        <div
          class="form-group"
        >
          <label>
            date of notification 
            <input
              class="form-control ng-untouched ng-pristine ng-valid"
              formcontrolname="dateNextNotification"
              ng-reflect-name="dateNextNotification"
              ng-reflect-required=""
              required=""
            />
          </label>
          
        </div>
        <div
          class="form-group"
        >
          <label>
            info 
            <input
              class="form-control ng-untouched ng-pristine ng-valid"
              formcontrolname="information"
              ng-reflect-name="information"
              ng-reflect-required=""
              required=""
            />
          </label>
          
        </div>
        <button
          class="btn btn-default"
          type="submit"
        >
          Submit
        </button>
        <button
          class="btn btn-default"
          type="button"
        >
          Reset
        </button>
      </form>
    </div>
    <div
      hidden=""
    >
      <div
        class="row"
      >
        <div
          class="col-sm"
        >
          UserName
        </div>
        <div
          class="col-sm"
        >
          Desout
        </div>
      </div>
      <div
        class="row"
      >
        <div
          class="col-sm"
        >
          Age
        </div>
        <div
          class="col-sm"
        >
          18
        </div>
      </div>
      <div
        class="row"
      >
        <div
          class="col-sm"
        >
          Birthday
        </div>
        <div
          class="col-sm"
        >
          2018/11/11
        </div>
      </div>
      <div
        class="row"
      >
        <div
          class="col-sm"
        >
          Date of Login
        </div>
        <div
          class="col-sm"
        >
          2018/11/11
        </div>
      </div>
      <div
        class="row"
      >
        <div
          class="col-sm"
        >
          date of notification
        </div>
        <div
          class="col-sm"
        >
          2018/11/11
        </div>
      </div>
      <div
        class="row"
      >
        <div
          class="col-sm"
        >
          info
        </div>
        <div
          class="col-sm"
        >
          infoinfo
        </div>
      </div>
      <button
        class="btn btn-default"
        type="button"
      >
        Reset
      </button>
      <button
        class="btn btn-default"
        type="button"
      >
        Go back
      </button>
    </div>
  </div>
</div>
`;
